bindings_ffi/src/mls.rs

```diff
@@ -1854,22 +1876,19 @@ impl FfiConversation {
 
     pub fn conversation_message_disappearing_settings(
         &self,
-    ) -> Result<FfiMessageDisappearingSettings, GenericError> {
-        let provider = self.inner.mls_provider()?;
-        let group_message_expiration_settings = self
-            .inner
-            .conversation_message_disappearing_settings(&provider)?;
+    ) -> Result<Option<FfiMessageDisappearingSettings>, GenericError> {
+        let settings = self.inner.client.group_disappearing_settings(self.id())?;
 
-        Ok(FfiMessageDisappearingSettings::new(
-            group_message_expiration_settings.from_ns,
-            group_message_expiration_settings.in_ns,
-        ))
+        match settings {
+            Some(s) => Ok(Some(FfiMessageDisappearingSettings::from(s))),
+            None => Ok(None),
+        }
     }
 
     pub fn is_conversation_message_disappearing_enabled(&self) -> Result<bool, GenericError> {
         let settings = self.conversation_message_disappearing_settings()?;
 
-        Ok(settings.from_ns > 0 && settings.in_ns > 0)
+        Ok(matches!(settings, Some(s) if s.from_ns > 0 && s.in_ns > 0))
```

Comment:
This is syntax I'd expect to see in Python, not Rust.

Reply:
Will fix it. 😬

Reply:
Oh I don't mind. I just thought it was funny, first time I've seen that.